pkg/csi/cinder/openstack/openstack_volumes.go

```diff
@@ -82,8 +83,18 @@ func (os *OpenStack) ListVolumes(ctx context.Context, limit int, startingToken s
 	var nextPageToken string
 	var vols []volumes.Volume
 
-	opts := volumes.ListOpts{Limit: limit, Marker: startingToken}
 	mc := metrics.NewMetricContext("volume", "list")
+	opts := volumes.ListOpts{Limit: limit, Marker: startingToken}
+	if limit == 0 {
+		page, err := volumes.List(os.blockstorage, opts).AllPages(ctx)
```

Comment:
in this case there is no need to provide opts
```suggestion
		page, err := volumes.List(os.blockstorage, nil).AllPages(ctx)
```

Reply:
Should be addressed now